File: DTSparkCore/src/main/java/com/kevin/java/transformations/Distinct_RDD.java
package com.kevin.java.transformations;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.VoidFunction;

import java.util.Arrays;
import java.util.List;

/**
 * @author kevin
 * @version 1.0
 * @description     对原RDD进行去重做操作，返回不重复的成员
 * @createDate 2018/12/27
 */
public class Distinct_RDD {

    public static void main(String[] args) {

        // 1.创建SparkConf，设置作业名称和模式
        SparkConf conf = new SparkConf().setAppName("Distinct_RDD").setMaster("local");

        // 2.基于Sparkconf对象创建一个SparkContext上下文，它是通往集群的唯一通道，且在创建时会创建任务调度器
        JavaSparkContext sc = new JavaSparkContext(conf);

        List<String> list = Arrays.asList("Java", "scala", "Python", "Java");

        // 3.去重
        JavaRDD<String>  distinctRDD = sc.parallelize(list).distinct();
        distinctRDD.foreach(new VoidFunction<String>() {
            public void call(String s) throws Exception {
                System.out.println(s);
            }
        });

        // 4.关闭
        sc.close();

    }

}

File: DTSparkCore/src/main/scala/com/kevin/scala/transformations/Distinct_RDD.scala
package com.kevin.scala.transformations

import org.apache.spark.{SparkConf, SparkContext}

/**
  * 对原RDD进行去重做操作，返回不重复的成员
  */
object Distinct_RDD {

  def main(args: Array[String]): Unit = {
    // 1.创建SparkConf
    val conf = new SparkConf().setAppName("Distinct_RDD").setMaster("local")
    // 2.创建SparkContext
    val sc = new SparkContext(conf)
    // 3.创建集合数据
    val list = List("Java", "scala", "Python", "Java")
    // 4.parallelize将集合转成rdd,distinct去重,foreach遍历打印
    sc.parallelize(list).distinct().foreach(println(_))
    // 5.关闭sc
    sc.stop()

  }

}
